React/Recipes/Session Handling.md

```diff
@@ -1,67 +1,168 @@
-## Motivation
+## Session model
 
-In almost every application we create, there is some session. _What is saved in this session is not essential now._ The session needs to be accessible in our components and based on that data, we can restrict/allow users access to a page.
+Most of the application we build have some kind of a login and protected pages. To make this possible, we use sessions. Using Datx, let's create a session model class.
 
-## `useSession` hook
+```ts
+import { Model, prop } from 'datx';
+
+export class SessionModel extends Model {
+  public static type = 'session';
+}
+```
+
+Session model is not really helpful on it's own and because of that it usually has a relationship to the user model. Now our session model looks like this
+
+```ts
+import { Model, prop } from 'datx';
+import { UserModel } from 'models/UserModel';
+
+export class SessionModel extends Model {
+  public static type = 'session';
 
-To make a session accessible in the React component, we'll create a `useSession` hook that will expose a session.
+  @prop.toOne(UserModel)
+  public user: UserModel;
+}
+```
+
+## Network
+
+Now, when we have defined session model we can create basic fetchers for the session. We'll create three fetchers - `createSession`, `readSession` and `deleteSession`. We'll create those fetchers using Datx.
 
-### SWR example
+_Following fetchers assume that application is using cookies for authentication. To use token or something else, the following methods can easily be extended._
+
+```ts
+import { SessionModel } from 'models/SessionModel';
+
+/**
+  Creates and returns a session from an endpoint.
+  User model is included in the response.
+*/
+export async function createSession(datx, loginData) {
+  return fetch(SESSION_API_ENDPOINT, {
```

Comment:
Sessions are maybe a wrong example for it, but it would be useful to see how this fetcher would look like with jsonapi